openff/qcsubmit/results/filters.py

```diff
@@ -19,6 +24,112 @@ class ResultFilter(BaseModel, abc.ABC):
     a specific criterion.
     """
 
+    @classmethod
+    def _basic_record_to_molecule(cls, cmiles: str, record: ResultRecord) -> Molecule:
+        """Creates a molecule object from a result record, storing the records
+        conformer on the molecule.
+
+        Args:
+            cmiles: The CMILES associated with the record.
+            record: The record to map to an OpenFF molecule.
+
+        Returns:
+            The record mapped to an OpenFF molecule.
+        """
+
+        qc_molecule: QCMolecule = record.get_molecule()
+
+        molecule: Molecule = Molecule.from_mapped_smiles(
+            cmiles, allow_undefined_stereo=True
+        )
+
+        molecule.add_conformer(
+            np.array(qc_molecule.geometry, float).reshape(-1, 3) * unit.bohr
+        )
+
+        return molecule
+
+    @classmethod
+    def _optimization_record_to_molecule(
+        cls, cmiles: str, record: OptimizationRecord
+    ) -> Molecule:
+        """Creates a molecule object from an optimization record, storing the lowest
+        energy conformer on the molecule.
+
+        Args:
+            cmiles: The CMILES associated with the record.
+            record: The record to map to an OpenFF molecule.
+
+        Returns:
+            The record mapped to an OpenFF molecule.
+        """
+
+        qc_molecule: QCMolecule = record.get_final_molecule()
+
+        molecule: Molecule = Molecule.from_mapped_smiles(
+            cmiles, allow_undefined_stereo=True
+        )
+
+        molecule.add_conformer(
+            np.array(qc_molecule.geometry, float).reshape(-1, 3) * unit.bohr
+        )
+
+        return molecule
+
+    @classmethod
+    def _torsion_drive_record_to_molecule(
+        cls, cmiles: str, record: TorsionDriveRecord
+    ) -> Molecule:
+        """Creates a molecule object from an torsion drive record, storing the lowest
+        energy conformer at each grid angle on the molecule.
+
+        Args:
+            cmiles: The CMILES associated with the record.
+            record: The record to map to an OpenFF molecule.
+
+        Returns:
+            The record mapped to an OpenFF molecule.
+        """
+
+        molecule: Molecule = Molecule.from_mapped_smiles(
+            cmiles, allow_undefined_stereo=True
+        )
+        qc_molecule: QCMolecule
+
+        for qc_molecule in record.get_final_molecules().values():
```

Comment:
For the fitting, we need the molecules and the results and calling these functions on every record can be quite slow, maybe we should batch the calls, so collect together the ids of the molecules and results we want and query them in chunks?

Reply:
Yeah that's a really good point! I've reconsidered this PR and now gone with a fully cached solution. I think really this is the only clean way to ensure we don't hit these expensive query calls.